include/deal.II/base/polynomial.h

```diff
@@ -87,7 +95,8 @@ namespace Polynomials
      * Return the value of this polynomial at the given point.
      *
      * This function uses the Horner scheme for numerical stability of the
-     * evaluation.
+     * evaluation for polynomials in the coefficient form or the product of
+     * terms involving the roots if that representation is used.
```

Comment:
Since the first review, I added the sentence
```
In case a manipulation is done that changes the roots, the
representation is switched to the coefficient form.
```
in the class description which should explain a bit better what happens.